### PR TITLE
selftests: switch to Python f-strings

### PR DESCRIPTION
**Pull request by ana**
Switching `selftests/ `  to see everything goes fine before I start looking into` avocado/`

References: https://github.com/avocado-framework/avocado/issues/5219

### CODE REVIEW

**Review by beraldoleal — CHANGES_REQUESTED** [review]
Hi @ana, those changes LGTM, and are welcome, thanks.

I just have one thing that bothers me: the lines length. ;( I found a few cases this change is replacing with a line longer than 79 columns, not recommend by PEP8.

Although this isn't in our contributor guide I feel like we've been following this pattern for quite some time now, and I confess it bothers me a little if we loosen this limitation.

**Comment by richtja** [comment]
> Hi @ana, those changes LGTM, and are welcome, thanks.
> 
> I just have one thing that bothers me: the lines length. ;( I found a few cases this change is replacing with a line longer than 79 columns, not recommend by PEP8.
> 
> Although this isn't in our contributor guide I feel like we've been following this pattern for quite some time now, and I confess it bothers me a little if we remove this limitation.

@beraldoleal I agree with you, and I think this should be part of our contributor guide. Also, it looks like we are missing the whole [code style guide](https://avocado-framework.readthedocs.io/en/latest/guides/contributor/chapters/styleguides.html#code-style-guide).

**Comment by ana** [comment]
> Hi @ana, those changes LGTM, and are welcome, thanks.
> 
> I just have one thing that bothers me: the lines length. ;( I found a few cases this change is replacing with a line longer than 79 columns, not recommend by PEP8.
> 
> Although this isn't in our contributor guide I feel like we've been following this pattern for quite some time now, and I confess it bothers me a little if we remove this limitation.

Hi @beraldoleal
Thanks for looking at this. If you want to enforce the max line length, the best is you enable the corresponding pylint check C0301  ( https://github.com/avocado-framework/avocado/blob/master/.pylintrc#L84 ) and update the value at https://github.com/avocado-framework/avocado/blob/master/.pylintrc#L385
Like enabling this pylint check, it'll require some work.

While I think enabling that pylint check is a good idea, I don't think that change should be coupled with this one or block it.

**Comment by beraldoleal** [comment]
> > Hi @ana, those changes LGTM, and are welcome, thanks.
> > I just have one thing that bothers me: the lines length. ;( I found a few cases this change is replacing with a line longer than 79 columns, not recommend by PEP8.
> > Although this isn't in our contributor guide I feel like we've been following this pattern for quite some time now, and I confess it bothers me a little if we remove this limitation.
> 
> Hi @beraldoleal Thanks for looking at this. If you want to enforce the max line length, the best is you enable the corresponding pylint check C0301 ( https://github.com/avocado-framework/avocado/blob/master/.pylintrc#L84 ) and update the value at https://github.com/avocado-framework/avocado/blob/master/.pylintrc#L385 Like enabling this pylint check, it'll require some work.
> 
> While I think enabling that pylint check is a good idea, I don't think that change should be coupled with this one or block it.

I might be wrong, but the way I see this is: C0301 checks are disabled now because we would need some work in order to enable it (as you mention) but not because we are in favor of having lines > 79 (I saw multiple reviews where we asked to reduce the line number to keep this convention). The same way we are disabling W0703 and many others there. We should avoid them whenever possible to make the job of reducing this list easier instead of harder when someone decides to do it.

If accepting f-strings over 79 columns is a common-sense here, I will be glad to remove the blocker flag. However, because of the fuzzy convention we have now (as noticed by @richtja) I think this should be voted/discussed among the core devs first.

@clebergnu or any other dev, comments?

**Review by richtja — COMMENTED** [review]
Except the @beraldoleal  concerns and conflicts with 62d8ab41f9118e191645bc69f38c7c21b0aa42e6, this LGTM.

**Comment by ana** [comment]
@beraldoleal I just checked while filing issue https://github.com/avocado-framework/avocado/issues/5285 and I haven't introduce any new longer line that wasn't already there with this PR.

**Comment by ana** [comment]
@richtja Fixed the conflict, thanks for pointing it out!

File: selftests/functional/plugin/spawners/test_process.py [hunk 1]
```diff
@@ -20,12 +20,11 @@ def test_logdir_path(self):
         test = script.Script(os.path.join(self.tmpdir.name, "logdir_test.py"),
                              TEST_LOGDIR)
         test.save()
-        result = process.run("%s run --job-results-dir %s --disable-sysinfo "
-                             "--json - -- %s" % (AVOCADO, self.tmpdir.name,
-                                                 test))
+        result = process.run(f"{AVOCADO} run --job-results-dir {self.tmpdir.name}"
```
Comment (beraldoleal) [inline selftests/functional/plugin/spawners/test_process.py:23]: When I made my comment, I was talking about this one: 77 vs 82.

File: selftests/functional/plugin/test_assets.py [hunk 2]
```diff
@@ -75,14 +75,12 @@ def test_asset_fetch_success(self):
         test_file.write(test_content.encode())
         test_file.close()
 
-        expected_output = "Fetching assets from %s.\n" \
-            "  File hello-2.9.tar.gz fetched or already on cache.\n" \
-            % test_file.name
+        expected_output = (f"Fetching assets from {test_file.name}.\n"
+                           f"  File hello-2.9.tar.gz fetched or already on cache.\n")
```
Comment (beraldoleal) [inline selftests/functional/plugin/test_assets.py:79]: This one, 85.

File: selftests/functional/plugin/test_assets.py [hunk 3]
```diff
@@ -237,12 +219,10 @@ def test_asset_fetch_unsupported_class(self):
...
         test_file.close()
 
-        expected_stdout = "Fetching assets from %s.\n" % test_file.name
+        expected_stdout = f"Fetching assets from {test_file.name}.\n"
         expected_rc = exit_codes.AVOCADO_ALL_OK
 
-        cmd_line = "%s --config %s assets fetch %s " % (AVOCADO,
-                                                        self.config_file.name,
-                                                        test_file.name)
+        cmd_line = f"{AVOCADO} --config {self.config_file.name} assets fetch {test_file.name} "
```
Comment (beraldoleal) [inline selftests/functional/plugin/test_assets.py:225]: This one, 95.

**Comment by beraldoleal** [comment]
@ana I found a few, that I comment here. But probably there is more.

**Review by beraldoleal — CHANGES_REQUESTED** [review]

File: selftests/functional/plugin/spawners/test_process.py (hunk 1 above)
Comment (ana) [reply selftests/functional/plugin/spawners/test_process.py:23]: sigh, I was mistaken by the fact that we run pylint *twice*  with different options in `selftests/lint.sh `

**Comment by ana** [comment]
@beraldoleal @richtja I have fixed the long lines introduce by my commit and probably some extra ones. Could you please take a look?

**Review by richtja — APPROVED** [review]
LGTM, thanks

**Review by beraldoleal — APPROVED** [review]
Thanks @ana for the changes.

**Pull request merged**